rust/src/tx_builder.rs

```diff
@@ -175,16 +182,58 @@ pub struct TransactionBuilder {
     prefer_pure_change: bool,
 }
 
+#[wasm_bindgen]
+#[derive(Clone, Builder, Debug)]
+pub struct TransactionBuilderConfig {
+    linear_fee: fees::LinearFee,
+    pool_deposit: BigNum,      // protocol parameter
+    key_deposit: BigNum,       // protocol parameter
+    max_value_size: u32,       // protocol parameter
+    max_tx_size: u32,          // protocol parameter
+    coins_per_utxo_word: Coin, // protocol parameter
+}
+
 #[wasm_bindgen]
 impl TransactionBuilder {
+    pub fn new(cfg: &TransactionBuilderConfig) -> Self {
+        Self {
+            coins_per_utxo_word: cfg.coins_per_utxo_word.clone(),
+            key_deposit: cfg.key_deposit.clone(),
+            pool_deposit: cfg.pool_deposit.clone(),
+            max_value_size: cfg.max_value_size,
+            max_tx_size: cfg.max_tx_size,
+            fee_algo: cfg.linear_fee.clone(),
+            inputs: Vec::new(),
+            outputs: TransactionOutputs::new(),
+            fee: None,
+            ttl: None,
+            certs: None,
+            withdrawals: None,
+            auxiliary_data: None,
+            input_types: MockWitnessSet {
+                vkeys: BTreeSet::new(),
+                scripts: BTreeSet::new(),
+                bootstraps: BTreeSet::new(),
+            },
+            validity_start_interval: None,
+            mint: None,
+            inputs_auto_added: false,
+            prefer_pure_change: false,
```

Comment:
@ozgrakkurt ,would that be possible to also move this one parameter `prefer_pure_change` to the config struct but as otpional? Something like `Option<boolean>` , so that it can be specified, but if it's not - then `.build()` for config does not fail?

Reply:
yep, that is possible, doing that now.